src/PhpSpec/CodeAnalysis/TokenizedTypeHintRewriter.php

```diff
@@ -52,6 +52,10 @@ public function __construct(TypeHintIndex $typeHintIndex, NamespaceResolver $nam
             $this->typehintTokens[] = T_NAME_FULLY_QUALIFIED;
             $this->typehintTokens[] = T_NAME_QUALIFIED;
         }
+
+        if (\PHP_VERSION_ID >= 80100) {
+            $this->typehintTokens[] = T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG;
```

Comment:
if we need to add the `&` here, don't we also need to add the `|` for union types on PHP 8+ ?

Reply:
This is already handled in the line below, as `token_get_all` returns `|` instead of a three element array for this. Please, check the position 7 of the array in this output https://3v4l.org/vkq1E#v8.0.16

https://github.com/phpspec/phpspec/blob/ec649c1b79c7e6041fbefa56e8998260a85cd40c/src/PhpSpec/CodeAnalysis/TokenizedTypeHintRewriter.php#L205